### ISSUE TITLE
Google sign -in button - "Use another account" does not provide a call back or consent form

### ISSUE THREAD

**Issue by j2emanue** [firebase/FirebaseUI-Android#1297]
here are the steps to reproduce the issue:

1. click on google signin Button.
2. choose "use another account"
3. sign in with the new account and this should bring you back to the app.
4. app only shows dark overlay and consent form or callback is not working.

then if i touch the screen the dark overlay goes away, and if i touch facebook button i get a toast saying an error has occured. at this time if i press the google button again it automatically logs me in with the account. I tried a few apps like stackoverflow,traveloka, airBNB, my own simple app with a google SSO button and glassdoor,lazada, etc and a few others and all of them give an error after getting to option 3. but i'd expect facebook button not to give an error at least .

**Comment by samtstern** [comment]
Thanks for reporting.  The main place we're tracking this issue is here:
https://github.com/googlesamples/google-services/issues/358

This requires an update to the Google Play services APK on your phone.  Not something we can fix with an SDK bump.

**Comment by Xiaojuanmao** [comment]
Which google play service version fix this issues, in my pixel, I update to the latest version, the same problem happens. My google play service version is 12.6.73(090400-194189626)

**Comment by samtstern** [comment]
@Xiaojuanmao this issue is still not fixed.

**Comment by koushikchoudhury0** [comment]
I use play services 15.0.1 and this problem exists !

**Comment by shahankitt** [comment]
@samtstern : It would be great if you could provide an update on this issue. I'm baffled that Google is taking so long to fix such a high priority issue which is not allowing the first time users to use the app. I'm sure this is being looked into but at least ETA can help us manage the expectations and behavior of real paying users on our app.. Also it would be highly appreciated if you or your team has been able to come up with any workaround whatsoever to unblock the huge set of users affected by this.

**Comment by samtstern** [comment]
@shahankitt we have a fix coming but not a workaround that works.  As I said, this has to be rolled into a new Google Play services release.  It's a big shame, but releasing Google Play services means rolling out a new app to literally a billion devices (or more!) so it's not something that happens quickly at all.  I can't share a timeline right now.

**Comment by patelrohan22** [comment]
@samtstern  

Until we get release update from google, here is the another work around by which I got solution .

As we know onActivityResult() does not called only when user add new account, but onRestart() and start() method get called.

Below is the code to open google concent screen to picked up account or add new account.

Intent signInIntent = loginHandler.mGoogleSignInClient.getSignInIntent();
        activity.startActivityForResult(signInIntent, RC_GET_TOKEN);
Before calling above code I keep currently registered google accounts count. i.e 3 accounts registered in my device.

Once user complete add new account process, onStart() method will get call and here I checked again google account count by below code:

    Account[] googleAccounts = AccountManager.get(activity).getAccountsByType("com.google");

    if (googleAccounts != null && googleAccounts.length > 0) {
        if (lastGoogleAccounts == 0) {
            lastGoogleAccounts = googleAccounts.length;
        } else {
            if (googleAccounts.length > lastGoogleAccounts) {
             // do login
}

if new count is > last count, it means new account is registered and redirect to home screen.

Hope this will help you.
